assets/src/edit-story/app/media/media3p/providerConfiguration.js

```diff
@@ -0,0 +1,132 @@
+ */
+
+/**
+ * Internal dependencies
+ */
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+/**
+ * External dependencies
+ */
+import React from 'react';
+import styled from 'styled-components';
+import { rgba } from 'polished';
+import theme from '../../../theme';
+import { ProviderType } from '../providerType';
+import { ReactComponent as UnsplashLogoFull } from '../../../icons/unsplash_logo_full.svg';
+
+const AttributionPill = styled.div`
+  position: absolute;
+  left: 24px;
+  bottom: 10px;
+  border-radius: 100px;
+  padding: 5px 8px;
+  line-height: 16px;
+  display: flex;
+  flex-wrap: nowrap;
+  font-size: 12px;
+  color: ${theme.colors.fg.white};
+  background-color: ${rgba(theme.colors.bg.black, 0.7)};
+  cursor: pointer;
+`;
+
+const LOGO_STYLE = {
+  fill: theme.colors.fg.white,
+  marginLeft: '6px',
+  height: '14px',
+};
+
+const ContentType = {
+  IMAGE: 'image',
+  VIDEO: 'video',
+};
+
+/**
+ * @typedef ProviderConfiguration
+ * @property {string} displayName The display name of the provider.
+ * @property {ContentType[]} supportedContentTypes The supported content types by the provider.
+ * @property {boolean} supportsCategories Whether this provider supports filtering media by category.
+ * @property {boolean} requiresAuthorAttribution Whether this provider requires showing author
+ * attribution on each media element.
+ * @property {function(): React.Component} attributionBuilder A function that constructs an attribution
+ * React Component for this provider.
+ */
+
+function UnsplashAttribution() {
+  return (
+    <a
+      href={
+        'https://unsplash.com?utm_source=web_stories_wordpress&utm_medium=referral'
+      }
+      target={'_blank'}
+      rel={'noreferrer'}
+    >
+      <AttributionPill>
+        {__('Powered by', 'web-stories')}
+        <UnsplashLogoFull style={LOGO_STYLE} />
+      </AttributionPill>
+    </a>
+  );
+}
+
+function CoverrAttribution() {
+  return (
+    <a href={'https://coverr.co'} target={'_blank'} rel={'noreferrer'}>
+      <AttributionPill>
+        {__('Powered by', 'web-stories')}
+        <span style={LOGO_STYLE}>{'COVERR'}</span>
+      </AttributionPill>
+    </a>
+  );
+}
+
+const showCoverrTab = false;
+
+/**
+ *
+ * @type {Object.<string, ProviderConfiguration>}
+ */
+export const Providers = Object.fromEntries(
+  new Map(
+    [
+      [
+        ProviderType.UNSPLASH,
+        {
+          displayName: 'Unsplash',
+          supportedContentTypes: [ContentType.IMAGE],
+          supportsCategories: true,
+          requiresAuthorAttribution: true,
+          attributionBuilder: UnsplashAttribution,
+        },
+      ],
+      showCoverrTab
+        ? [
+            ProviderType.COVERR,
+            {
+              displayName: 'Coverr',
+              supportedContentTypes: [ContentType.VIDEO],
+              supportsCategories: false,
+              requiresAuthorAttribution: false,
+              attributionBuilder: CoverrAttribution,
```

Comment:
Why did you not just put the properties for the attribution here instead of the whole component?

Reply:
Because we might need different styles of attribution per provider. Maybe Unsplash is ok with a "Powered by <SVG>" but Tenor wants something different.

Reply:
That makes sense. But we can define what needs to be customized and provided. It does not just need to be one image or so.

Reply:
I think I'll leave this as is, and in the meantime will confirm with Tenor and Unsplash if this is the attribution format they both expect.